lib/puppet/provider/pulp_rpmbind/consumer.rb

```diff
@@ -46,12 +46,26 @@ def exists?
   end
 
   def create
-    consumer('rpm', 'bind', '--repo-id', @resource[:name])
+    output = consumer('rpm', 'bind', '--repo-id', @resource[:name])
+    raise "Repository [#{@resource[:name]}] does not exist on the server" if output =~ /does not exist/
+    wait_for_bind # 'pulp-consumer rpm bind' happens asynchronously in the background
     @property_hash[:ensure] = :present
   end
 
   def destroy
     consumer('rpm', 'unbind', '--repo-id', @resource[:name])
     @property_hash[:ensure] = :absent
   end
+
+  def wait_for_bind
+    tries ||= 10
```

Comment:
I wonder if this should be configurable. What's your experience with how long the waiting typically takes?

Reply:
During my testing, the process was actually very quick.  One 250ms wait was *always* enough for me. (IIRC I had to set the sleep to 0.01 to actually see more than one retry take place).  But I figured it might not always be that fast depending on what else the system was doing at the time or because of network congestion etc.  Waiting for no more than 2.5 seconds total seemed reasonable?  If it doesn't complete by then, there's probably no point waiting any longer??

Reply:
That's fair and we can always fix it later.